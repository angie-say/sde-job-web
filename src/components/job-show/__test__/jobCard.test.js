import React from 'react';     // general config antd
import ReactDOM from "react-dom";   
// import puppeteer from 'puppeteer'

import ReactTestRenderer from 'react-test-renderer';
import {render, cleanup} from '@testing-library/react'
import JobCard from '../jobCard';

afterEach(cleanup);
it('renders correctly', () => {
    const tree = ReactTestRenderer.create(<JobCard 
            id={1}
            title={"test"}
            isLike={true}
            changeLike={()=>{}}
            company={"test company"}
            cities={"test city"}
            from_url={"www.google.com"}
            apply_url={""}
            publish_time={1649027672}
            content={"test description"}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


describe('Login', () => {
  let browser
  let page

//   beforeAll(async () => {
//     browser = await puppeteer.launch({ args: ['--no-sandbox'] })
//   })

  beforeEach(async () => {
    page = await browser.newPage()
    await page.goto('http://localhost:8000/en/login', {
      waitUntil: 'networkidle2',
    })
  })

  afterEach(() => page.close())

  it('should login with failure', async () => {
    await page.waitFor(selector => !!document.querySelector('#username'), {
      timeout: 3000,
    })
    await page.type('#username', 'wrong_user')
    await page.type('#password', 'wrong_password')
    await page.click('button[type="button"]')
    await page.waitForSelector('.anticon-close-circle') // should display error
  })

  it('should login successfully', async () => {
    await page.waitForSelector('#username', { timeout: 3000 })
    await page.type('#username', 'admin')
    await page.type('#password', 'admin')
    await page.click('button[type="button"]')
    await page.waitForSelector('.ant-layout-footer')
    const text = await page.evaluate(() => document.body.innerHTML)
    expect(text).toContain('Ant Design Admin')
  })

  afterAll(() => browser.close())
})
