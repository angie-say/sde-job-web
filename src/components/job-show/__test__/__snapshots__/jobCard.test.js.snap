// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      class="job-card-head"
    >
      <div
        class="job-card-head-link"
      >
        <a
          className="job-card"
          href="www.google.com"
          target="_blank"
        >
          <div
            className="job-card-head-link-title"
          >
            test
          </div>
        </a>
        <div
          class="job-card-head-link-operation"
        >
          <span
            aria-label="heart"
            className="anticon anticon-heart"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="img"
            style={
              Object {
                "color": "#F00",
              }
            }
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              data-icon="heart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
              />
            </svg>
          </span>
          <a
            href=""
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            target="_blank"
          >
            <span
              aria-label="link"
              className="anticon anticon-link"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="link"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M574 665.4a8.03 8.03 0 00-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 00-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 000 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 000 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 00-11.3 0L372.3 598.7a8.03 8.03 0 000 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <div
        class="job-card-head-info"
      >
        <div
          class="job-card-head-info-layer1"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
            style={Object {}}
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <span
                  aria-label="bank"
                  className="anticon anticon-bank"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="bank"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                    />
                  </svg>
                </span>
                test company
              </div>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <div
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <span
                  aria-label="environment"
                  className="anticon anticon-environment"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="environment"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                    />
                  </svg>
                </span>
                test city
              </div>
            </div>
          </div>
        </div>
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            aria-label="field-time"
            className="anticon anticon-field-time"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="field-time"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M945 412H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h256c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM811 548H689c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h122c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM477.3 322.5H434c-6.2 0-11.2 5-11.2 11.2v248c0 3.6 1.7 6.9 4.6 9l148.9 108.6c5 3.6 12 2.6 15.6-2.4l25.7-35.1v-.1c3.6-5 2.5-12-2.5-15.6l-126.7-91.6V333.7c.1-6.2-5-11.2-11.1-11.2z"
              />
              <path
                d="M804.8 673.9H747c-5.6 0-10.9 2.9-13.9 7.7a321 321 0 01-44.5 55.7 317.17 317.17 0 01-101.3 68.3c-39.3 16.6-81 25-124 25-43.1 0-84.8-8.4-124-25-37.9-16-72-39-101.3-68.3s-52.3-63.4-68.3-101.3c-16.6-39.2-25-80.9-25-124 0-43.1 8.4-84.7 25-124 16-37.9 39-72 68.3-101.3 29.3-29.3 63.4-52.3 101.3-68.3 39.2-16.6 81-25 124-25 43.1 0 84.8 8.4 124 25 37.9 16 72 39 101.3 68.3a321 321 0 0144.5 55.7c3 4.8 8.3 7.7 13.9 7.7h57.8c6.9 0 11.3-7.2 8.2-13.3-65.2-129.7-197.4-214-345-215.7-216.1-2.7-395.6 174.2-396 390.1C71.6 727.5 246.9 903 463.2 903c149.5 0 283.9-84.6 349.8-215.8a9.18 9.18 0 00-8.2-13.3z"
              />
            </svg>
          </span>
          4/3/2022
        </div>
      </div>
    </div>
    <a
      className="job-card"
      href="www.google.com"
      target="_blank"
    >
      <div
        className="job-card-content"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "test description",
            }
          }
        />
      </div>
    </a>
  </div>
</div>
`;
